gobblin-utility/src/main/java/org/apache/gobblin/broker/SharedResourcesBrokerFactory.java

```diff
@@ -78,13 +84,32 @@
     return new SharedResourcesBrokerImpl<>(new DefaultBrokerCache<S>(),
         scopeWrapper,
         Lists.newArrayList(new SharedResourcesBrokerImpl.ScopedConfig<>(globalScope.getType(),
-            ConfigUtils.getConfigOrEmpty(addSystemConfigurationToConfig(config), BrokerConstants.GOBBLIN_BROKER_CONFIG_PREFIX))),
+            getBrokerConfig(addSystemConfigurationToConfig(config)))),
         ImmutableMap.of(globalScope.getType(), scopeWrapper));
   }
 
   private static InheritableThreadLocal<SharedResourcesBroker<?>> threadLocalBroker = new ThreadLocalBroker();
   private static SharedResourcesBroker<SimpleScopeType> SINGLETON;
 
+  /**
+   * Get all broker configurations from the given {@code srcConfig}. Configurations are loaded in order of
+   * namespaces, which is encoded as a comma separated string keyed by
+   * {@value BrokerConstants#GOBBLIN_BROKER_CONFIG_NAMESPACES}. The default namespaces is just
+   * {@value BrokerConstants#GOBBLIN_BROKER_CONFIG_PREFIX}
+   */
+  @VisibleForTesting
+  static Config getBrokerConfig(Config srcConfig) {
```

Comment:
We should probably always load `gobblin.broker` namespace, and just allow adding additional namespaces.